src/components/ExcalidrawView.tsx

```diff
@@ -13,65 +12,172 @@ import {
   ExcalidrawImperativeAPI,
 } from '@excalidraw/excalidraw/types/types';
 
-import { DEBOUNCE_VALUE } from '../config/settings';
-import getInitialData from './InitialData';
+import {
+  APP_DATA_TYPES,
+  ExcalidrawElementsAppData,
+  ExcalidrawStateAppData,
+} from '../config/appDataTypes';
+import {
+  DEFAULT_EXCALIDRAW_ELEMENTS_APP_DATA,
+  DEFAULT_EXCALIDRAW_STATE_APP_DATA,
+} from '../config/constants';
+import {
+  DEBOUNCE_COMPARE_ELEMENTS,
+  DEBOUNCE_SAVE_ELEMENTS,
+  DEBOUNCE_SAVE_STATE,
+} from '../config/settings';
+import { reconcileElements } from '../utils/reconciliation';
 import { useAppDataContext } from './context/AppDataContext';
 
-type Props = {
-  appData: AppData;
-};
-
-const ExcalidrawView: FC<Props> = ({ appData }) => {
+const ExcalidrawView: FC = () => {
   const excalidrawRef = useRef<ExcalidrawImperativeAPI>(null);
   const viewModeEnabled = false;
   const zenModeEnabled = false;
   const gridModeEnabled = false;
   const theme = 'light';
-  const { patchAppData } = useAppDataContext();
-  // eslint-disable-next-line react/destructuring-assignment
-  const { id, data } = appData;
-  const iData = getInitialData(
-    data.elements as readonly ExcalidrawElement[],
-    // data.state as AppState,
-  );
+  const date = new Date();
+  const name = `drawing${date.toISOString()}`;
+
+  const [localElements, setLocalElements] =
+    useState<readonly ExcalidrawElement[]>();
```

Comment:
Is this state supposed to be readonly ? In that case, you don't need the `setLocalElements` ? 
```ts
const [localElements] = useState<readonly ExcalidrawElement[]>();
```